### ISSUE TITLE
ARIMA forecasting backtest

### ISSUE THREAD

**Issue by jinbo-huang** [statsmodels/statsmodels#8770]
Hello everyone,

I'm new to time series analysis. I'm currently trying to analyze a price series with ARMA and ARIMA (trying to reproduce the result of "Machine Trading" from Ernest Chan). The goal is to extract the parameters from fitted models and use those parameters in C++ to predict future mid points. In the example below, I tried to use the parameters from the fitted ARMA/ARIMA model to do some predictions on the test set, but I don't know how to compute the prediction manually. I trained a ARIMA(2, 0, 5) model and extracted the parameters from it. However, I cannot reproduce the values came from `forecast()` and `predict()`.

![image](https://user-images.githubusercontent.com/4936182/229750322-478be33f-9622-4277-8987-64743eb97c82.png)

Here's my code
```
# Split the data into training and testing data 
# mid_points is the array containing historical mid points
train_data: npt.NDArray = mid_points[len(mid_points)-252*(24*60-15) * 2:len(mid_points)-252*(24*60-15)] 
test_data: npt.NDArray = mid_points[len(mid_points)-252*(24*60-15):]

# Train models
model = ARIMA(train_data, order=(2, 0, 5), trend="c")
fit_result = model.fit()

# One-step prediction (forecast, predict)
print(fit_result.predict(start=len(train_data), end=len(train_data), dynamic=True))
print(fit_result.forecast(1))
# [0.92627827]
# [0.92627827]

# One-step prediction (apply)
new_res = fit_result.apply(mid_points[len(train_data) - best_q:len(train_data)])
new_res.forecast(1)
# array([0.97994715])

# One-step prediction (manual)
intercept = fit_result.params[0] * (1 - np.sum(fit_result.arparams))
ar_part = np.dot(fit_result.arparams, mid_points[len(train_data) - 2:len(train_data)][::-1])
ma_part = np.dot(fit_result.maparams, residuals[-len(ma_params):][::-1])
result = intercept + ar_part + ma_part
# result = 0.9800231921389104
```

As you can see in the above code, the value of manual prediction is quite different from `forecast()` and `predict()`. I have checked [link1](https://machinelearningmastery.com/make-manual-predictions-arima-models-python/), [link2](https://stackoverflow.com/questions/63383393/why-statsmodels-arima1-0-0-is-not-equivalent-to-autoreg1) , #8535 and traced old `_get_predict_out_of_sample()`, but I still failed to reproduce the result. Is there any code file that I can trace to reproduce the result from ARMA and ARIMA?
